.github/workflows/validate.yml

```diff
@@ -0,0 +1,61 @@
+name: CI
+on:
+  push:
+    branches: [ main ]
+    tags: ['*']
+
+  pull_request:
+
+permissions:
+  contents: write
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Checkout Course Management Tools Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: 11
+          distribution: adopt-hotspot
+          cache: 'sbt'
+
+      - name: Setup Coursier
+        run: |
+          curl -fLo coursier https://git.io/coursier-cli &&
+          chmod +x coursier &&
+          ./coursier
+
+      - name: Check code formatting
+        run: sbt "scalafmtCheckAll ; scalafmtSbtCheck"
```

Comment:
Total preference, but I typically actually like to put this in it's own `formatting` job. Then you can still see if your tests pass even if you have a formatting issue. Currently your tests won't run here if your stuff isn't formatted. Not a huge deal, just a preference.

Reply:
If you were to do this, you could remove this and then just add the following to the bottom:

```yml
  format-check:
    runs-on: ubuntu-latest

    steps:

      - name: Checkout Course Management Tools Repo
        uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: 11
          distribution: adopt-hotspot
          cache: 'sbt'

      - name: Check code formatting
        run: sbt "scalafmtCheckAll ; scalafmtSbtCheck"
```

Reply:
Good suggestion. Will do this in a second step (reason being that I haven't found out what makes some tests fail on Windows on some PR's. So, doing this in smaller steps)

Reply:
Btw I'd still recommend doing this before merging. Main reason is that now you're actually checking formatting on both runs which isn't necessary.